htdocs/js/ui_utils.js

```diff
@@ -182,6 +183,41 @@ ui_utils.install_common_ace_key_bindings = function(widget, get_language) {
                         shell.scroll_to_end();
                     });
             }
+        },
+
+        {
+            name: 'execute-cursor-at-beginning',
+            bindKey: {
+          
+                mac: 'Ctrl-A',
+                sender: 'editor'
+            },
+            exec: function(widget, args, request) {
+                if (widget.getOption("readOnly"))
+                    return;
+                //row of the cursor on current line
+                var row = widget.getCursorPosition().row;
+                //move to the beginning of that line
+                widget.navigateTo(row, 0);
+                //make sure it appears at beginning of text
+                widget.navigateLineStart();
+            }
+        } ,
+        {
+            name: 'execute-cursor-at-end',
+            bindKey: {
+       
+                mac: 'Ctrl-E',
+                sender: 'editor'
+            },
+            exec: function(widget, args, request) {
+                //row of the cursor on current line
+                var row = widget.getCursorPosition().row;
+                //last column of the cursor on current line
+                var lastCol = widget.getSession().getValue().length;
```

Comment:
I think it returns the value of the current cell, i haven't gotten infinity when testing with my code. I'll look into it though